### PR TITLE
QA Transformer

### PR DESCRIPTION
**Pull request by fabiopetroni**
Transformer model for Question Answering (span extraction).

## Example usage:

Fine-tune the bert-large-uncased-whole-word-masking model on SQuAD v1.1:
```bash
python examples/train_model.py -m transformer_qa -t squad -mf /tmp/transformer_qa -bs 8 -tr 384 -lr 3e-5 --model_type bert --model_name_or_path bert-large-uncased-whole-word-masking --do_lower_case True -eps 2 --t-total 21900
```

test:{'exs': 10570, 'accuracy': 0.7991, 'f1': 0.8838, 'bleu': 0.1761, 'lr': 0, 'num_updates': 5475, 'examples': 175200, 'loss': 3135.0, 'mean_loss': 0.01789}

### CODE REVIEW

**Comment by fabiopetroni** [comment]
After implementing the WarmupLinearSchedule I could reach 88.38 f1-score (+1.5 from reduceonplateau), still far from 93.15 f1 score claimed in here [https://github.com/huggingface/pytorch-transformers]. Unfortunately, in order to mimic exactly the approach in the huggingface script I need to be able to control the construction of the batches from my transformer_qa class - i.e., I need to delete and add examples dynamically. @stephenroller do you have any suggestion on how can I do that?

**Comment by stephenroller** [comment]
Abandoning, @hadasah is implementing the LinearScheduler soon, so we can revisit this in 2020 if we want.
